core/cmd/client.go

```diff
@@ -50,11 +51,10 @@ import (
 )
 
 var prometheus *ginprom.Prometheus
+var initOnce sync.Once
 
-func init() {
-	// ensure metrics are registered once per instance to avoid registering
-	// metrics multiple times (panic)
-	prometheus = ginprom.New(ginprom.Namespace("service"))
+func initPrometheus(cfg config.GeneralConfig) {
+	prometheus = ginprom.New(ginprom.Namespace("service"), ginprom.Token(cfg.PrometheusAuthToken()))
```

Comment:
Will this still work if there is no token?  (Or will tokens be mandatory now?)

Reply:
The token defaults to "", and the default behavior of ginprom on "" is to drop authentication